txp/Pos/Txp/Logic/Local.hs

```diff
@@ -48,18 +48,16 @@ type TxpLocalWorkMode ctx m =
     , HasLens GenesisUtxo ctx GenesisUtxo
     )
 
--- CHECK: @processTx
--- #processTxDo
+-- | Process transaction. 'TxId' is expected to be the hash of
+-- transaction in 'TxAux'. Separation is supported for optimization
+-- only.
 txProcessTransaction
     :: TxpLocalWorkMode ctx m
     => (TxId, TxAux) -> ExceptT ToilVerFailure m ()
 txProcessTransaction itw@(txId, txAux) = do
     let UnsafeTx {..} = taTx txAux
     tipDB <- GS.getTip
-    epoch <- siEpoch <$>
-        -- TODO: Don't use inaccurate slotting here. If we don't know the
-        -- current slot, we should reject transactions. See CSL-1341.
-        getCurrentSlotInaccurate
+    epoch <- maybe (throwError ToilSlotUnknown) (pure . siEpoch) =<< getCurrentSlot
```

Comment:
maybe it's not important for you, but important for managers (gromak, georgee, and I wonder because don't want to go out context). also, it might be unimportant for you for now, but may be important in the future when you take a related task, and if a related discussion is in the private chats you aren't able to read them